core/controllers/editor.py

```diff
@@ -758,3 +760,65 @@ def get(self, exploration_id):
         self.render_json({
             'unresolved_answers': unresolved_answers_with_frequency
         })
+
+
+class LearnerAnswerInfoHandler(EditorHandler):
+    """Handles the learner answer info for an exploration state."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_edit_entity
+    def get(self, entity_type, entity_id):
+        """Handles the GET requests for learner answer info for an
+        exploration state.
+        """
+        if not constants.ENABLE_SOLICIT_ANSWER_DETAILS_FEATURE:
+            raise self.PageNotFoundException
+
+        if entity_type == feconf.ENTITY_TYPE_EXPLORATION:
+            state_name = self.request.get('state_name')
+            if not state_name:
+                raise self.InvalidInputException
+            state_reference = (
+                stats_services.get_state_reference_for_exploration(
+                    entity_id, state_name))
+        elif entity_type == feconf.ENTITY_TYPE_QUESTION:
+            state_reference = (
+                stats_services.get_state_reference_for_question(
+                    entity_id))
+
+        learner_answer_details = stats_services.get_learner_answer_details(
+            entity_type, state_reference)
+        learner_answer_info_dict_list = []
+        if learner_answer_details is not None:
+            learner_answer_info_dict_list = [
+                learner_answer_info.to_dict() for learner_answer_info in
+                learner_answer_details.learner_answer_info_list]
+        self.render_json({
+            'learner_answer_info_dict_list': learner_answer_info_dict_list
+        })
+
+    @acl_decorators.can_edit_entity
+    def delete(self, entity_type, entity_id):
+        """Deletes the learner answer info by the given id."""
+
+        if not constants.ENABLE_SOLICIT_ANSWER_DETAILS_FEATURE:
+            raise self.PageNotFoundException
+
+        if entity_type == feconf.ENTITY_TYPE_EXPLORATION:
+            state_name = self.request.get('state_name')
+            if not state_name:
+                raise self.InvalidInputException
+            state_reference = (
+                stats_services.get_state_reference_for_exploration(
+                    entity_id, state_name))
+        elif entity_type == feconf.ENTITY_TYPE_QUESTION:
+            state_reference = (
+                stats_services.get_state_reference_for_question(
+                    entity_id))
+        learner_answer_info_id = self.request.get('learner_answer_info_id')
```

Comment:
What if the entity type is neither exploration or question? should we throw an error?